app/models/miq_queue.rb

```diff
@@ -237,7 +237,12 @@ def self.put_or_update(find_options)
     #   it's unordered nature.
     where_scope = if conds.key?(:args)
                     args = YAML.dump conds.delete(:args)
-                    MiqQueue.where(conds).where(['args = ?', args])
```

Comment:
Lol, I just posted something like this https://github.com/ManageIQ/manageiq/issues/5965 "put_unless_exists and args"

We are close to removing `args_selector`. The next step after that is to remove `args` from the criteria for a unique message.

Having said that, there are some cases where args does make the message unique, and we have to change that code before modifying this.